go-controller/pkg/kubevirt/dhcp.go

```diff
@@ -0,0 +1,144 @@
+package kubevirt
+
+import (
+	"fmt"
+	"net"
+
+	corev1 "k8s.io/api/core/v1"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	utilnet "k8s.io/utils/net"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+const (
+	dhcpLeaseTime = 3500
+)
+
+type dhcpConfigs struct {
+	V4 *nbdb.DHCPOptions
+	V6 *nbdb.DHCPOptions
+}
+
+func EnsureDHCPOptionsForVM(controllerName string, nbClient libovsdbclient.Client, watchFactory *factory.WatchFactory, pod *corev1.Pod, ovnPodAnnotation *util.PodAnnotation, lsp *nbdb.LogicalSwitchPort) error {
```

Comment:
AFAIU, you only need the IPs from the ovn annotations; why not pass the `[]*net.IPNet` directly ? 

Furthermore, you can compute the VM name from outside, using the `kubevirt.ExtractVMNameFromPod` function. This would allow you to use `NamespacedName`s on the `EnsureDHCPOptionsForVM` and `composeDHCPConfigs` functions. This way, you don't need to pass the pod to a function named `...ForVM` .